dashboard/app/controllers/pd/mimeo_sso_controller.rb

```diff
@@ -0,0 +1,76 @@
+# Authenticate and redirect a Code Studio user to Mimeo using their single-sign-on (SSO) API
+class Pd::MimeoSsoController < ApplicationController
+  load_resource :enrollment, class: 'Pd::Enrollment', find_by: :code, id_param: :enrollment_code
+
+  ENDPOINT = 'https://my.mimeo.com/sso/authenticate.ashx'.freeze
+  AUTHORIZED_MARKETPLACE_URL = 'unpluggedlessonmaterials'.freeze
+  SSO_VERSION = '2.0'.freeze
+
+  ALLOWED_COURSES = [Pd::Workshop::COURSE_CSF].freeze
+
+  def authenticate_and_redirect
+    # Make sure the user is logged in and is the owner of this enrollment
+    authorize! :manage, @enrollment
+    raise CanCan::AccessDenied unless ALLOWED_COURSES.include? @enrollment.workshop.course
+
+    unless @enrollment.completed_survey?
+      # Until the survey is processed, it won't show up in the enrollment model.
+      # Check pegasus forms directly before giving up.
+      raise CanCan::AccessDenied unless PEGASUS_DB[:forms].where(kind: 'PdWorkshopSurvey', source_id: @enrollment.id).any?
+    end
+
+    # Set up fields for the Mimeo SSO API. Each of these will be rendered as a hidden field on a form which will be
+    # automatically posted to the @endpoint on page load.
+    # Mimeo's API will then authenticate and either redirect the user's browser to the redirectUrl on success,
+    # or to our errorUrl if it fails (see #error below).
+    @endpoint = ENDPOINT
+    @fields = {
+      firstName: encrypt_token(@enrollment.first_name),
+      lastName: encrypt_token(@enrollment.last_name),
+      email: encrypt_token(current_user.email),
```

Comment:
We do [require](https://github.com/code-dot-org/code-dot-org/blob/a2adb2db73b4a17d6cbc5d4cfd9145eae13ba8e1/dashboard/app/models/pd/enrollment.rb#L37-L41) first and last names for new workshop enrollments. It is true that some old enrollments might have no last name. These should have already taken the survey, but anyway it's possible. 

I just verified that a nil or empty last name causes an error:
![image](https://cloud.githubusercontent.com/assets/8405316/22757654/96c8a1c8-ee00-11e6-90fc-501be03557de.png)

So I'll go ahead and substitute a placeholder `-` in the case of blank last name.